monai/handlers/segmentation_saver.py

```diff
@@ -88,24 +91,24 @@ def _create_file_basename(postfix, input_file_name, folder_path, data_root_dir="
 
     def __call__(self, engine):
         """
-        This method assumes:
-            - 3rd output of engine.state.batch is a meta data dict, and have the keys:
+        This method assumes to have the keys:
             'filename_or_obj' -- for output file name creation
             and optionally 'original_affine', 'affine' for data orientation handling.
             - output file datatype from `engine.state.output.dtype`.
         """
+        meta_data = self.batch_transform(engine.state.batch)
         meta_data = engine.state.batch[2]  # assuming 3rd output of input dataset is a meta data dict
```

Comment:
need to remove line 100

Reply:
could have this docstring: this method assumes `self.batch_transform` will extract meta data from the input batch, that is, `meta_data = self.batch_transform(engine.state.batch)`

Reply:
fixed in the latest patch.
Thanks.